code/modules/atmospherics/environmental/LINDA_fire.dm

```diff
@@ -136,6 +141,16 @@
 		COMSIG_ATOM_ABSTRACT_ENTERED = PROC_REF(on_entered),
 	)
 	AddElement(/datum/element/connect_loc, loc_connections)
+	var/turf/open/our_turf = loc
+	//on creation we check adjacent turfs for hot spot to start grouping, each hotspot should begin with a hot group with themself in it
+	our_hot_group = new
+	our_hot_group.add_to_group(src)
+	for(var/turf/open/to_check as anything in our_turf.atmos_adjacent_turfs)
+		if(to_check.active_hotspot)
+			var/obj/effect/hotspot/enemy_spot = to_check.active_hotspot
+			if(enemy_spot.our_hot_group == our_hot_group)
+				break
+			our_hot_group.merge_hot_groups(enemy_spot.our_hot_group)//we keep merging with surrounding hot groups till its a soup of spots
```

Comment:
You create a group, add a hotspot to it and then look for another group to merge your new one with. Looking for a new group first and then if that fails, making a new one would be better. Also, did you mean for this to be a break and not a continue? This way you're relying entirely on the order of atmos_adjacent_turfs to merge groups, you can end up with two unmerged groups next to eachother